### ISSUE TITLE
Add the ability to combine `Query`s  into composite Queries

### ISSUE THREAD

**Issue by favilo** [sminez/penrose#294]
### Please describe the change / addition you'd like to see made

I'd like to see an ergonomic interface that allows me to combine `Query` structs in simple ways.

Specifically I'd like to write something like the following:
```rust
    let manage_hook = manage_hooks! {
		ClassName("zoom").and(
            Titles(ZOOM_TILE_TITLES.to_vec()))
        => DefaultTiled,
        ClassName("zoom").and(
            Titles(ZOOM_TILE_TITLES.to_vec()).not()
        ) => FloatingRelative(top_right_corner),
        ClassName("obsidian").and(
            Titles(vec!["Obsidian Help"]))
         => FloatingCentered::new(0.25, 0.5),
	}
```

And have it work in the generally expected way.

- Is this completely new functionality, or an enhancement to existing functionality?

This is just an enhancement to what already exists.

- What kind of feature is it that you are requesting?
  - [ ] Changes to core
  - [ ] A new layout
  - [ ] A hook implementation
  - [ ] A custom extension
  - [X] New X interactions
  - [ ] Other

### Is this a feature you have seen in other window managers?

If so, please provide any links or reference to the existing implementation
and some details on how the feature is used.

Yes, I've used these types of Queries with XMonad.

I'm extending my own personal penrose config based on this article: https://www.peterstuart.org/posts/2021-09-06-xmonad-zoom/.

**Comment by favilo** [comment]
I've submitted #291, as a first stab at this.

**Comment by sminez** [comment]
Hi @favilo, I'm not sure if you saw the comments I left on your PR? Taking a quick look at [your config](https://github.com/favilo/favilrose/blob/main/src/hooks.rs#L134-L150) it looks like the class name check can just be made part of the custom `Titles` query that you've already set up? While I _do_ like the idea of being able to compose together hooks in this way, there are a couple of issues with what you have provided in the PR. If this is still a feature that you'd like to see added to Penrose then please do let me know if what I've commented on the PR is all making sense. If not then I'll close this as `won't fix` until there is someone who is wanting to actively make use of this as a feature so I can be sure that the implementation satisfies their use case :slightly_smiling_face:

**Comment by favilo** [comment]
Ah, sorry, I've been rather busy this last month, so I haven't had time to work on the PR, but yes, I saw the comments.

I would really like the ability to combine the Queries, and I think your suggestions are valid. Give me a few days, and I'll hopefully find some time to tidy up the PR.

**Comment by sminez** [comment]
Implemented in #291
